src/Tizen.Multimedia.Remoting/WebRTC/MediaPacketSource.cs

```diff
@@ -0,0 +1,242 @@
+            {
+                Log.Error(WebRTCLog.Tag, "The format of the packet is invalid : Unmatched format.");
+                throw new ArgumentException("The format of the packet is invalid : Unmatched format.");
+            }
+
+            if (packet.Format.Type == MediaFormatType.Video && _videoMediaFormat == null)
+            {
+                Log.Error(WebRTCLog.Tag, "Video is not configured with the current source.");
+                throw new ArgumentException("Video is not configured with the current source.");
+            }
+
+            if (packet.Format.Type == MediaFormatType.Audio && _audioMediaFormat == null)
+            {
+                Log.Error(WebRTCLog.Tag, "Audio is not configured with the current source.");
+                throw new ArgumentException("Audio is not configured with the current source.");
+            }
+
+            WebRtc.ValidateWebRTCState(WebRTCState.Negotiating, WebRTCState.Playing);
+
+            NativeWebRTC.PushMediaPacket(WebRtc.Handle, packet.GetHandle()).
+                ThrowIfFailed("Failed to push the packet to the WebRTC");
+        }
+
+        private void SetMediaStreamInfo(MediaFormat mediaFormat)
+        {
+            if (mediaFormat == null)
+            {
+                Log.Warn(WebRTCLog.Tag, "invalid media format");
+            }
+            else
+            {
+                IntPtr ptr = IntPtr.Zero;
+
+                try
+                {
+                    ptr = mediaFormat.AsNativeHandle();
+                    NativeWebRTC.SetMediaPacketSourceInfo(WebRtc.Handle, SourceId.Value, ptr).
+                        ThrowIfFailed("Failed to set the media stream info");
+                }
+                finally
+                {
+                    MediaFormat.ReleaseNativeHandle(ptr);
+                }
+            }
+        }
+
+        internal override void OnAttached(WebRTC webRtc)
+        {
+            Debug.Assert(webRtc != null);
+
+            if (WebRtc != null)
+            {
+                Log.Error(WebRTCLog.Tag, "The source is has already been assigned to another WebRTC.");
+                throw new InvalidOperationException("The source is has already been assigned to another WebRTC.");
+            }
+
+            NativeWebRTC.AddMediaSource(webRtc.Handle, MediaSourceType.MediaPacket, out uint sourceId).
+                ThrowIfFailed("Failed to add media source for media packet.");
+
+            WebRtc = webRtc;
+            SourceId = sourceId;
+
+            AudioConfiguration?.OnWebRTCSet();
+            VideoConfiguration?.OnWebRTCSet();
+
+            SetMediaStreamInfo(_audioMediaFormat);
+            SetMediaStreamInfo(_videoMediaFormat);
+        }
+
+        internal override void OnDetached(WebRTC webRtc)
+        {
+            NativeWebRTC.RemoveMediaSource(webRtc.Handle, SourceId.Value).
+                ThrowIfFailed("Failed to add media source for media packet.");
```

Comment:
typo here -> Failed to 'remove' media source ...

Reply:
Done.